home/models.py

```diff
@@ -340,6 +341,29 @@ def whatsapp_template_name(self):
     def whatsapp_template_body(self):
         return self.whatsapp_body.raw_data[0]["value"]["message"]
 
+    def clean(self):
+        message_with_media_length = 1024
+        errors = []
+        for message in self.whatsapp_body:
+            if (
+                (
+                    "image" in message.value
+                    and "document" in message.value
+                    and "media" in message.value
+                )
+                and (
+                    message.value["image"]
+                    or message.value["document"]
+                    or message.value["media"]
+                )
+                and len(message.value["message"]) > message_with_media_length
+            ):
+                errors.append(
+                    f"A WhatsApp message with media cannot be longer than {message_with_media_length} characters long, your message is {len(message.value['message'])} characters long"
+                )
+        if errors:
+            raise ValidationError(errors)
```

Comment:
Should we maybe have a unit test, to ensure that the error gets raised when we expect it to?

Reply:
yes we should probably have unit tests for everything :D